### PR TITLE
Fix `sigma_noise` gradient propagation in `BaseLaplace.log_likelihood`

### PR DESCRIPTION
**Pull request by kourbou**
Commit 1eea27c introduces type hints in `baselaplace.py` but breaks the gradient flow through `BaseLaplace.log_likelihood(...)` since `sigma_noise` loses its gradient via [`torch.tensor(...)`]. 

This is fixed by using [`torch.as_tensor(...)`] as it converts floats but preserves the gradient. (See highlighted note [here](https://pytorch.org/docs/stable/generated/torch.tensor.html#:~:text=preserves%20autograd%20history).)

This change fixes the `UserWarning` below and passes type checking with Pylance.
```
laplace/baselaplace.py:226: UserWarning: To copy construct from a tensor, it is recommended to use
[...], rather than torch.tensor(sourceTensor).
```

[`torch.tensor(...)`]: https://pytorch.org/docs/stable/generated/torch.tensor.html
[`torch.as_tensor(...)`]: https://pytorch.org/docs/stable/generated/torch.as_tensor.html

### CODE REVIEW

**Comment by aleximmer** [comment]
Thanks a lot for bisecting the issue and fixing it as well.
